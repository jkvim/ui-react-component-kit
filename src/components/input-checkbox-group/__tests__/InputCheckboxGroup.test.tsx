import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { InputCheckboxGroup } from '../';

describe('InputCheckboxGroup', () => {

  const options = [ { label: 'label', value: 'value' }, { label: 'one', value: '1' } ];
  const values = [ 'value' ];

  describe('props', () => {

    const mockCallback = jest.fn();

    const wrapper = mount(
      <InputCheckboxGroup
        id="id"
        name="name"
        options={options}
        values={values}
        alignment="horizontal"
        breakpoint="md"
        hint="hint"
        errorMessage="error"
        className="class"
        disabled={true}
        onChange={mockCallback}
      />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `name` in props', () => {
      expect(wrapper.props().name).toBe('name');
    });

    it('should pass `options` in props', () => {
      expect(wrapper.props().options).toEqual(options);
    });

    it('should pass `values` in props', () => {
      expect(wrapper.props().values).toEqual(values);
    });

    it('should pass `hint` in props', () => {
      expect(wrapper.props().hint).toBe('hint');
    });

    it('should pass `alignment` in props', () => {
      expect(wrapper.props().alignment).toBe('horizontal');
    });

    it('should pass `breakpoint` in props', () => {
      expect(wrapper.props().breakpoint).toBe('md');
    });

    it('should pass `errorMessage` in props', () => {
      expect(wrapper.props().errorMessage).toBe('error');
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });

    it('should pass `disabled` in props', () => {
      expect(wrapper.props().disabled).toBe(true);
    });

    it('should pass `onChange` in props', () => {
      expect(wrapper.props().onChange).toBe(mockCallback);
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<InputCheckboxGroup id="id" name="name" options={options} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot error', () => {
    const wrapper = mount(<InputCheckboxGroup id="id" name="name" options={options} errorMessage="Error" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot disabled', () => {
    const wrapper = mount(<InputCheckboxGroup id="id" name="name" options={options} disabled={true} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should pre-select radio', () => {
    const wrapper = mount(<InputCheckboxGroup id="id" name="name" options={options}  values={['value', '1']} />);
    const inputs = wrapper.find('input');

    expect(inputs.get(0).props.checked).toBe(true);
    expect(inputs.get(1).props.checked).toBe(true);
  });

  it('expect value to change when keys are pressed', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<InputCheckboxGroup id="id" name="name" options={options} onChange={mockCallback} />);
    wrapper.find('input').at(1).simulate('change', { target: { value:  '1'}});
    // The mock function is called once
    expect(mockCallback).toBeCalled();
    // The new value was sent
    expect(mockCallback.mock.calls[0][0]).toEqual(['1']);
  });

  it('expect everything to work when no onChange is provided', () => {
    const wrapper = mount(<InputCheckboxGroup id="id" name="name" options={options} />);
    wrapper.find('input').at(1).simulate('change', { target: { value: '1' } });
  });
});
