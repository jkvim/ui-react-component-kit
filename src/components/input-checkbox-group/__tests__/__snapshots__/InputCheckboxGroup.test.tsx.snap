// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheckboxGroup should match snapshot 1`] = `
<InputCheckbox
  id="testId"
  name="testName"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
      Object {
        "label": "Three",
        "value": "3",
      },
    ]
  }
  values={
    Array [
      "2",
      "3",
    ]
  }
>
  <styled.div>
    <div
      className="sc-htoDjs gQgvQJ"
    >
      <styled.fieldset
        alignment="horizontal"
        aria-describedby="testId-error"
      >
        <fieldset
          aria-describedby="testId-error"
          className="sc-dnqmqq dcDEKm"
        >
          <styled.div
            alignment="horizontal"
            key="0"
          >
            <div
              className="sc-jzJRlG jNxNtV"
            >
              <styled.input
                aria-invalid={false}
                checked={false}
                hasError={false}
                id="testId-0"
                innerRef={[Function]}
                name="testName"
                onChange={[Function]}
                theme={
                  Object {
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#2fa4e7",
                      "checked": "#2fa4e7",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "hover": "#178acc",
                    },
                    "font": Object {
                      "size": "1.8rem",
                    },
                  }
                }
                type="checkbox"
                value="1"
              >
                <input
                  aria-invalid={false}
                  checked={false}
                  className="sc-fjdhpX dKHXyf"
                  id="testId-0"
                  name="testName"
                  onChange={[Function]}
                  type="checkbox"
                  value="1"
                />
              </styled.input>
              <styled.label
                hasError={false}
                htmlFor="testId-0"
                theme={
                  Object {
                    "color": Object {
                      "border": "#2fa4e7",
                      "error": "#c71c22",
                      "font": "#333333",
                    },
                    "font": Object {
                      "size": "1.8rem",
                    },
                  }
                }
              >
                <label
                  className="sc-bwzfXH dXCFSf"
                  htmlFor="testId-0"
                >
                  One
                </label>
              </styled.label>
            </div>
          </styled.div>
          <styled.div
            alignment="horizontal"
            key="1"
          >
            <div
              className="sc-jzJRlG jNxNtV"
            >
              <styled.input
                aria-invalid={false}
                checked={true}
                hasError={false}
                id="testId-1"
                innerRef={[Function]}
                name="testName"
                onChange={[Function]}
                theme={
                  Object {
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#2fa4e7",
                      "checked": "#2fa4e7",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "hover": "#178acc",
                    },
                    "font": Object {
                      "size": "1.8rem",
                    },
                  }
                }
                type="checkbox"
                value="2"
              >
                <input
                  aria-invalid={false}
                  checked={true}
                  className="sc-fjdhpX dKHXyf"
                  id="testId-1"
                  name="testName"
                  onChange={[Function]}
                  type="checkbox"
                  value="2"
                />
              </styled.input>
              <styled.label
                hasError={false}
                htmlFor="testId-1"
                theme={
                  Object {
                    "color": Object {
                      "border": "#2fa4e7",
                      "error": "#c71c22",
                      "font": "#333333",
                    },
                    "font": Object {
                      "size": "1.8rem",
                    },
                  }
                }
              >
                <label
                  className="sc-bwzfXH dXCFSf"
                  htmlFor="testId-1"
                >
                  Two
                </label>
              </styled.label>
            </div>
          </styled.div>
          <styled.div
            alignment="horizontal"
            key="2"
          >
            <div
              className="sc-jzJRlG jNxNtV"
            >
              <styled.input
                aria-invalid={false}
                checked={true}
                hasError={false}
                id="testId-2"
                innerRef={[Function]}
                name="testName"
                onChange={[Function]}
                theme={
                  Object {
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#2fa4e7",
                      "checked": "#2fa4e7",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "hover": "#178acc",
                    },
                    "font": Object {
                      "size": "1.8rem",
                    },
                  }
                }
                type="checkbox"
                value="3"
              >
                <input
                  aria-invalid={false}
                  checked={true}
                  className="sc-fjdhpX dKHXyf"
                  id="testId-2"
                  name="testName"
                  onChange={[Function]}
                  type="checkbox"
                  value="3"
                />
              </styled.input>
              <styled.label
                hasError={false}
                htmlFor="testId-2"
                theme={
                  Object {
                    "color": Object {
                      "border": "#2fa4e7",
                      "error": "#c71c22",
                      "font": "#333333",
                    },
                    "font": Object {
                      "size": "1.8rem",
                    },
                  }
                }
              >
                <label
                  className="sc-bwzfXH dXCFSf"
                  htmlFor="testId-2"
                >
                  Three
                </label>
              </styled.label>
            </div>
          </styled.div>
        </fieldset>
      </styled.fieldset>
      <styled.div
        hasError={false}
        id="testId-error"
        theme={
          Object {
            "color": Object {
              "error": "#c71c22",
            },
            "font": Object {
              "size": "1.8rem",
            },
          }
        }
      >
        <div
          className="sc-VigVT gUinmJ"
          id="testId-error"
        />
      </styled.div>
    </div>
  </styled.div>
</InputCheckbox>
`;
